src/Functions/array/indexOf.cpp

```diff
@@ -5,12 +5,30 @@ namespace DB
 {
 struct NameIndexOf { static constexpr auto name = "indexOf"; };
 
-/// indexOf(arr, x) - returns the index of the element x (starting with 1), if it exists in the array, or 0 if it
-/// doesn't.
 using FunctionIndexOf = FunctionArrayIndex<IndexOfAction, NameIndexOf>;
 
 REGISTER_FUNCTION(IndexOf)
 {
-    factory.registerFunction<FunctionIndexOf>();
+    FunctionDocumentation::Description description = R"(
+Returns the index of the first element with value 'x' (starting from 1) if it is in the array.
+If the array does not contain the searched-for value, the function returns `0`.
+
+Elements set to `NULL` are handled as normal values.
+    )";
+    FunctionDocumentation::Syntax syntax = "indexOf(arr, x)";
+    FunctionDocumentation::Arguments arguments = {
+        {"arr", "An array to search in for `x`. [`Array`](/sql-reference/data-types/array)."},
+        {"x", "Value of the first matching element in `arr` for which to return an index. UInt64."},
+    };
+    FunctionDocumentation::ReturnedValue returned_value = "Returns the index (numbered from one) of the first `x` in `arr` if it exists. Otherwise, returns `0`.";
+    FunctionDocumentation::Examples examples = {
+        {"Basic example", "SELECT indexOf([1, 3, 3, 4], 3)", "2"},
```

Comment:
Lets use an unsorted array as input, we don't want the reader to think that sorting is somehow mandatory.